libs/features/mas/web-components/src/hydrate.js

```diff
@@ -12,29 +12,33 @@ export async function hydrate(fragmentData, merchCard) {
     if (!variant) return;
     fragment.model = fragment.model;
 
-    merchCard.variantLayout?.refs?.forEach((ref) => ref.remove());
+    // remove all previous slotted content except the default slot
+    merchCard.querySelectorAll('[slot]').forEach((el) => {
+        el.remove();
+    });
     merchCard.variant = variant;
     await merchCard.updateComplete;
-    merchCard.variantLayout.refs ??= [];
+
     const { aemFragmentMapping } = merchCard.variantLayout;
 
     if (!aemFragmentMapping) return;
 
     const appendFn = (el) => {
-        merchCard.variantLayout.refs.push(el);
         merchCard.append(el);
     };
 
-    fragment.mnemonicIcon?.forEach((icon, idx) => {
-        const href =
-            fragment.mnemonicLink?.length > idx
-                ? fragment.mnemonicLink[idx]
-                : '';
-        const alt =
-            fragment.mnemonicAlt?.length > idx ? fragment.mnemonicAlt[idx] : '';
+    const mnemonics = fragment.mnemonicIcon?.map((icon, index) => ({
+        icon,
+        alt: fragment.mnemonicAlt[index] ?? '',
+        link: fragment.mnemonicLink[index] ?? '',
+    }));
+
+    fragmentData.computed = { mnemonics };
```

Comment:
As we design mnemonic icon, alt text and link as 3 multi fields, these values need to be pre-processed before mapping them to fields in Studio.

Reply:
imo we should have no trace of Studio related logic in Milo, why not doing it in Studio when the edit window is opened?
there is a risk of someone removing it in the future because it looks in milo like it's not used

Reply:
@3ch023 since MAS changes already have been merged, I raised https://jira.corp.adobe.com/browse/MWPW-160932 to track this.
I propose to address it post PR merge.

Reply:
sure np with that